app/models/agents/csv_agent.rb

```diff
@@ -0,0 +1,195 @@
+module Agents
+  class CsvAgent < Agent
+    include FormConfigurable
+    include FileHandling
+
+    cannot_be_scheduled!
+    consumes_file_pointer!
+
+    def default_options
+      {
+        'mode' => 'parse',
+        'separator' => ',',
+        'use_fields' => '',
+        'output' => 'event_per_row',
+        'with_header' => 'true',
+        'data_path' => '$.data',
+        'data_key' => 'data'
+      }
+    end
+
+    description do
+      <<-MD
+        The `CsvAgent` parses or serializes CSV data. When parsing, events can either be emitted for the entire CSV, or one per row.
+
+        Set `mode` to `parse` to parse CSV from incoming event, when set to `serialize` the agent serilizes the data of events to CSV.
+
+        ### Universal options
+
+        Specify the `separator` which is used to seperate the fields from each other (default is `,`).
+
+        `data_key` sets in the key which contains the serialized CSV or parsed CSV data in both the emitted and incoming events
+
+        ### Parsing
+
+        If `use_fields` is set to a comma seperated string and the CSV file contains field headers the agent will only extract the specified fields.
+
+        `output` determines wheather one event per row is emitted or one event that includes all the rows.
+
+        Set `with_header` to `true` if first line of the CSV file are field names.
+
+        #{receiving_file_handling_agent_description}
```

Comment:
When no file pointer is received it will look for data in `data_path`, see the description below:
"When receiving the CSV data in a regular event use [JSONPath](http://goessner.net/articles/JsonPath/) to select the path in `data_path`. `data_path` is only used when the received event does not contain a 'file pointer'."